benchmark/cmake/modules/AddSwiftBenchmarkSuite.cmake

```diff
@@ -267,6 +267,33 @@ function(add_swift_multisource_nonwmo_benchmark_library objfiles_out)
   set(${objfiles_out} ${objfiles} PARENT_SCOPE)
 endfunction()
 
+function(swift_benchmark_suite_version)
```

Comment:
This is the core of my question from OP. When this is called from `swift_benchmark_compile` it gets called multiple time, which is wrong. I can call it from the `configure_build` macro, to run only once. But that is also incorrect, because manual edits of the `Version.swift` source file get picked up into the benchmark binary but don't trigger cmake file re-generation.

When I manually delete the generated `Version.swift` file, I get `ninja: error: '/Users/mondo/Developer/swift-source/swift/benchmark/utils/Version.swift', needed by 'benchmark/Onone-x86_64--macosx10.9/DriverUtils.o', missing and no known rule to make it`

[Googling for that error](https://www.google.com/search?q=ninja%3A+error%3A+*+needed+by+*+missing+and+no+known+rule+to+make+it) seems to point to in the direction of a [workaround](https://github.com/ninja-build/ninja/issues/720#issuecomment-37243765)(?), but I don't understand cmake and ninja enough to know how... Do you know how to make a "rule to make it"?

Reply:
I was thinking about this. My suggestion:

1. Why not name the non-configure version Version.swift.
2. Why not name the configure version CMakeVersion.swift.in.
3. I would configure this separately from the actual build commands, i.e. I would do it in the top level file. Then I would add special code in swift_compile_benchmark specifically to compile that code in the build directory location.

Reply:
This is the first time I'm working with cmake. The few descriptions of generating git based versions using cmake look reasonable:
* http://xit0.org/2013/04/cmake-use-git-branch-and-commit-details-in-project/
* https://bravenewmethod.com/2017/07/14/git-revision-as-compiler-definition-in-build-with-cmake/
But I have no idea how to correctly integrate this into the quite complicated Swift's build system with many includes. One issue is we want this to be executed on every build, not just in the configure phase. From what I've read so far, I have a vague hunch the generated `CMakeVersion.swift` should be in it's own target, but I have no idea how to accomplish that yet.